docs/docs/noir/standard_library/meta/expr.md

```diff
@@ -161,3 +161,21 @@ comptime {
 #include_code is_continue noir_stdlib/src/meta/expr.nr rust
 
 `true` if this expression is `continue`.
+
+### map
+
+#include_code map noir_stdlib/src/meta/expr.nr rust
+
+Applies a mapping function to this expression or to all of its sub-expressions.
+
+If `f` returns `Option::some`, that's the return value of this method. Otherwise,
+`f` will be recursively called on each of this expression sub-expressions.
+
+For example, calling `map` on `[1, 2, [3, 4]]` with an `f` that returns `Option::some`
+for expressions that are integers, doubling them, would return `[2, 4, [6, 8]]`.
```

Comment:
```suggestion
Applies a mapping function to this expression and to all of its sub-expressions.

`f` will be applied to each sub-expression first, then applied to the expression itself.
This happens recursively for every expression within `self`.

For example, calling `map` on `(&[1], &[2, 3])` with an `f` that returns `Option::some`
for expressions that are integers, doubling them, would return `(&[2], &[4, 6])`.
```
- Slight wording change to fit the new suggestion I made for the method
- Changed the example a bit to be a valid value